[Merged by Bors] -  chore(*): adaptations for batteries#1545

Vierkantor opened the pull request:
After [batteries#1545](https://github.com/leanprover-community/batteries/pull/1545) is merged:

- [x] Edit the lakefile to point to leanprover-community/batteries:main
- [x] Run lake update batteries
- [x] Merge leanprover-community/mathlib4:master
- [ ] Wait for CI and merge

---
<!-- Your PR title will become the first line of the commit message.

In this box, the text above the `---` (if not empty) will be appended
to the commit message, and can be used to give additional context or
details. Please leave a blank newline before the `---`, otherwise GitHub
will format the text above it as a title.

For details on the "pull request lifecycle" in mathlib, please see:
https://leanprover-community.github.io/contribute/index.html

In particular, note that most reviewers will only notice your PR
if it passes the continuous integration checks.
Please ask for help on https://leanprover.zulipchat.com if needed.

When merging, all the commits will be squashed into a single commit
listing all co-authors.

Co-authors in the squash commit are gathered from two sources:

First, all authors of commits to this PR branch are included. Thus,
one way to add co-authors is to include at least one commit authored by
each co-author among the commits in the pull request. If necessary, you
may create empty commits to indicate co-authorship, using commands like so:

git commit --author="Author Name <author@email.com>" --allow-empty -m "add Author Name as coauthor"

Second, co-authors can also be listed in lines at the very bottom of
the commit message (that is, directly before the `---`) using the following format:

Co-authored-by: Author Name <author@email.com>

If you are moving or deleting declarations, please include these lines
at the bottom of the commit message (before the `---`, and also before
any "Co-authored-by" lines) using the following format:

Moves:
- Vector.* -> List.Vector.*
- ...

Deletions:
- Nat.bit1_add_bit1
- ...

Any other comments you want to keep out of the PR commit should go
below the `---`, and placed outside this HTML comment, or else they
will be invisible to reviewers.

If this PR depends on other PRs, please list them below this comment,
using the following format:
- [ ] depends on: #abc [optional extra text]
- [ ] depends on: #xyz [optional extra text]

-->

[![Open in Gitpod](https://gitpod.io/button/open-in-gitpod.svg)](https://gitpod.io/from-referrer/)

Commit:
Update Batteries branch for testing https://github.com/leanprover-community/batteries/pull/1545

Vierkantor committed:
Remove duplicate `library_note2` command.

fgdorais committed:
Merge remote-tracking branch 'upstream/master' into batteries-pr-testing-1545

Vierkantor committed:
Fix

Vierkantor committed:
This library note got upstreamed to Batteries; we can delete it here.

fgdorais committed:
Merge remote-tracking branch 'upstream/master' into batteries-pr-testing-1545

fgdorais committed:
chore: update batteries

grunweg commented on Mathlib/Tactic/Basic.lean:
```diff
@@ -12,6 +12,7 @@ public meta import Mathlib.Tactic.Lemma
 public meta import Mathlib.Tactic.TypeStar
 public meta import Mathlib.Tactic.Linter.OldObtain
 public meta import Mathlib.Tactic.Simproc.ExistsAndEq
+public import Batteries.Util.LibraryNote -- For `library_note` command.
```
Seems good!

grunweg reviewed (commented):
Looks good to me - with the ext note moved.
bors d+

Vierkantor committed:
Move the library note to a new file.

Still importing `library_note` in `Tactic.Basic`; we can decide to shake this later but for now I think it's useful to make the command available throughout Mathlib.

Vierkantor commented:
bors r+

Vierkantor committed:
`mk_all`
